File: src/main/java/net/miarma/mkernel/commands/admin/HealCommand.java
package net.miarma.mkernel.commands.admin;

import java.util.List;

import org.bukkit.Bukkit;
import org.bukkit.entity.Player;

import dev.jorel.commandapi.CommandAPICommand;
import net.miarma.mkernel.config.CommandWrapper;
import net.miarma.mkernel.config.providers.CommandProvider;
import net.miarma.mkernel.config.providers.MessageProvider;
import net.miarma.mkernel.util.MessageUtil;

public class HealCommand {
    public static void register() {
        CommandWrapper healCmd = CommandProvider.getHealCommand();
        new CommandAPICommand(healCmd.getName())
                .withOptionalArguments(
                	CommandProvider.Arguments.playersOptArg().withPermission(healCmd.getPermission().others())
                )
                .withPermission(healCmd.getPermission().base())
                .withFullDescription(healCmd.getDescription())
                .withShortDescription(healCmd.getDescription())
                .withUsage(healCmd.getUsage())
            .executesPlayer((sender, args) -> {
                if(args.count() == 0) {
                    sender.setHealth(20);
                    sender.setFoodLevel(20);
                    MessageUtil.sendMessage(sender, healCmd.getMessages()[0], true);
                }

                if(args.count() == 1) {
                    Player target = Bukkit.getPlayer(args.getRaw(0));

                    if(target == null) {
                        MessageUtil.sendMessage(sender, MessageProvider.Errors.playerNotFound(), true);
                    }

                    target.setHealth(20);
                    target.setFoodLevel(20);

                    MessageUtil.sendMessage(sender, healCmd.getMessages()[1], true,
                            List.of("%player%"), List.of(target.getName()));
                    MessageUtil.sendMessage(target, healCmd.getMessages()[2], true,
                            List.of("%sender%"), List.of(sender.getName()));
                }
            })
            .register();
    }
}

File: src/main/kotlin/net/miarma/mkernel/command/admin/HealCommand.kt
package net.miarma.mkernel.command.admin

import com.google.inject.Inject
import com.google.inject.Singleton
import dev.jorel.commandapi.arguments.PlayerProfileArgument
import dev.jorel.commandapi.kotlindsl.commandAPICommand
import dev.jorel.commandapi.kotlindsl.playerExecutor
import net.miarma.mkernel.common.service.impl.ConfigService
import net.miarma.mkernel.common.service.impl.MessageService
import net.miarma.mkernel.util.PlayerUtil

@Singleton
class HealCommand @Inject constructor(
    private val configService: ConfigService,
    private val messageService: MessageService
) {
    fun register() {
        commandAPICommand(configService.getString("commands.heal.name")) {
            withPermission(configService.getString("commands.heal.permissions.base"))
            withOptionalArguments(
                PlayerProfileArgument(configService.getString("arguments.player"))
                    .withPermission(configService.getString("commands.heal.permissions.others"))
            )
            withFullDescription(configService.getString("commands.heal.description"))
            withUsage(configService.getString("commands.heal.usage"))
            playerExecutor { sender, args ->
                if (args[0] == null) {
                    sender.foodLevel = 20
                    sender.health = 20.0
                    sender.fireTicks = 0
                    messageService.builder(configService.getString("commands.heal.messages.healedSelf")).withPrefix().send(sender)
                    return@playerExecutor
                }

                val target = PlayerUtil.fromArg(args[0])
                if (target == null || !target.isOnline) {
                    messageService.builder(configService.getString("language.errors.playerNotFound")).withPrefix().send(sender)
                    return@playerExecutor
                }

                target.foodLevel = 20
                target.health = 20.0
                target.fireTicks = 0

                messageService.builder(configService.getString("commands.heal.messages.healedPlayer")).withPrefix().tag("player", target.name).send(sender)
                messageService.builder(configService.getString("commands.heal.messages.beenHealed")).withPrefix().forPlayer(target).tag("sender", sender.name).send(target)
            }
        }
    }
}